cmd/kubeadm/app/cmd/phases/kubelet.go

```diff
@@ -96,7 +97,8 @@ func runKubeletStart(c workflow.RunData) error {
 	// Write env file with flags for the kubelet to use. We do not need to write the --register-with-taints for the master,
 	// as we handle that ourselves in the markmaster phase
 	// TODO: Maybe we want to do that some time in the future, in order to remove some logic from the markmaster phase?
-	if err := kubeletphase.WriteKubeletDynamicEnvFile(&data.Cfg().NodeRegistration, data.Cfg().FeatureGates, false, data.KubeletDir()); err != nil {
+	pauseImage := images.GetPauseImage(data.Cfg().ClusterConfiguration.ImageRepository)
+	if err := kubeletphase.WriteKubeletDynamicEnvFile(&data.Cfg().NodeRegistration, data.Cfg().FeatureGates, pauseImage, false, data.KubeletDir()); err != nil {
```

Comment:
something that @rosti suggested was suggested in the original PR was that we should pass ClusterConfiguration instead of feature gates and pause image to WriteKubeletDynamicEnvFile:
https://github.com/kubernetes/kubernetes/pull/66798/files#r206508620

Reply:
In fact, we can replace the first three parameters with `InitConfiguration` here and move the `GetPauseImage` call to `WriteKubeletDynamicEnvFile`. This will have the benefit of tidying up the code and removing duplicate logic between here and `postupgrade.go`